### ISSUE TITLE
Оffline mode

### ISSUE THREAD

**Issue by noneme** [rasmuslos/ShelfPlayer#11]
```
func getAudiobookTrackUrl(trackId: String) -> URL {
        documentsURL.appending(path: "tracks").appending(path: "\(trackId).flac")
    }
    
    func deleteEpisode(episodeId: String) {
        try? FileManager.default.removeItem(at: getEpisodeUrl(episodeId: episodeId))
    }
    
    func getEpisodeUrl(episodeId: String) -> URL {
        documentsURL.appending(path: "tracks").appending(path: "\(episodeId).E.flac")
    }
```

Why does the file format change to *.flac when downloading books/podcasts?

This is not a native format for Apple, which means that some books (I have not yet understood the pattern) cannot be played offline after downloading.
Moreover, if you change the file extension in the code when saving to native *.m4a, then everything is okay and all books are played perfectly

**Comment by tehguitarist** [comment]
.m4b should also be entirely fine (which is likely more common than .m4a for audiobooks).

**Comment by rasmuslos** [comment]
This is actually a pretty stupid hack to make `AVPlayer` play downloaded tracks. It does not want to play files without an extension but it works fine when it the extension is wrong. Or so I thought. I will look into that and do some testing with `m4b` files, but fixing this could take some time

**Comment by noneme** [comment]
Thank you! the fact that any modification requires some time for analysis and implementation is normal and correct! especially since this is not a critical moment and may have low priority.

**Comment by rasmuslos** [comment]
It seems like using the wrong extension is fine as long as the file type is roughly the same, I guess m4a uses a different encoder then flac/mp3/... I have to rewrite some parts of the download logic to fix this, but thanks for pointing this out!

**Comment by tehguitarist** [comment]
I don’t suppose there’s a way to pull extension information when downloading the file is there?

**Comment by rasmuslos** [comment]
I am currently restructuring `ShelfPlayerKit` which includes reworking the download logic to some extend, so I will store the file extension in the database which should solve this issue

**Comment by rasmuslos** [comment]
> I don’t suppose there’s a way to pull extension information when downloading the file is there?

Its not that hard, I was doing this because it is way easier and the code is less complex, but because it does not work I will do just that

**Comment by rasmuslos** [comment]
This should be fixed in the latest release `v2.0.7` (https://github.com/rasmuslos/ShelfPlayer/commit/cc34fd4c1a32a13ec4e12137e510ea08a7aee451)
